src/topology/unit_interval.lean

```diff
@@ -80,3 +81,52 @@ meta def unit_interval : tactic unit :=
 `[apply unit_interval.le_one] <|> `[apply unit_interval.one_minus_le_one]
 
 end tactic.interactive
+
+
+section
+variables {𝕜 : Type*} [field 𝕜] [topological_space 𝕜] [topological_ring 𝕜]
+
+/--
+The map `λ x, a * x + b`, as a homeomorphism from `𝕜` (a topological field) to itself, when `a ≠ 0`.
+-/
+@[simps]
+def affine_homeomorph (a b : 𝕜) (h : a ≠ 0) : 𝕜 ≃ₜ 𝕜 :=
+{ to_fun := λ x, a * x + b,
+  inv_fun := λ y, (y - b) / a,
+  left_inv := λ x, by { simp only [add_sub_cancel], exact mul_div_cancel_left x h, },
+  right_inv := λ y, by { simp [mul_div_cancel' _ h], }, }
+
+end
+
+section
+variables {𝕜 : Type*} [linear_ordered_field 𝕜] [topological_space 𝕜] [topological_ring 𝕜]
+
+/--
+The image of `[0,1]` under the homeomorphism `λ x, a * x + b` is `[b, a+b]`.
+-/
+-- We only need the ordering on `𝕜` here to avoid talking about flipping the interval over.
+-- At the end of the day I only care about `ℝ`, so I'm hesitant to put work into generalizing.
+lemma affine_homeomorph_image_I (a b : 𝕜) (h : 0 < a) (w) :
```

Comment:
Calling the assumption `w` feels weird here.

Reply:
Also, please give the type of `w` explicitly. Lean might not need it, but for a human readability I think it is important.

Reply:
Even better, I removed it entirely.